### DISCUSSION TITLE
3D boolean of two meshes. Don't understand.

### DISCUSSION THREAD

**Discussion by ghost** [CGAL/cgal#6319] [Q&A] [answered]
Hi,
I would like to understand the library better.
Even after reading [3d boolean](https://doc.cgal.org/latest/Nef_3/index.html#title5) 
I still cannot get through it.

What if I need a cube cut by a cylinder? So it would result in a cube with two faces having a hole.

How to achieve this basic? 
It would be nice to have a high level function for union, difference.. like in blender for example.
Thanks

**Comment by sloriot** [comment, marked as answer]
For most case, using the newer corefinement API is enough and simpler (while being faster and as robust). See https://doc.cgal.org/latest/Polygon_mesh_processing/index.html#Coref_section

Input are triangle meshes. In your case, it would be a call to [`CGAL::Polygon_mesh_processing::corefine_and_compute_difference(cube, cylinder, output)`](https://doc.cgal.org/latest/Polygon_mesh_processing/group__PMP__corefinement__grp.html#gac5a853c33c4b0a9da9403c9b191caa44)

**Reply by ghost** [reply]
Ah it seems like it. I will do that. Thank you!
